ulog/log_test.go

```diff
@@ -34,6 +34,16 @@ import (
 	"github.com/uber-go/zap"
 )
 
+func TestZapSink(t *testing.T) {
+	l, s := TestingLogger()
+	l.Info("This is my message", "And this is", "My field")
+	assert.Equal(t, 1, len(s.Logs()))
+
+	e := s.Logs()[0]
+	assert.Equal(t, "This is my message", e.Msg)
+	assert.Contains(t, e.Fields, zap.String("And this is", "My field"))
```

Comment:
Certainly some tests I have in mind in the future, we can see if some tests can be improved by this approach.
I think it's not worth it to go change all the tests that use `WithInMemoryLogger` to this.